zerver/migrations/0709_fix_topics_for_direct_messages.py

```diff
@@ -0,0 +1,50 @@
+# Generated by Django 5.1.8 on 2025-05-23 09:25
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Max
+
+
+def fix_topic_for_direct_messages(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    Message = apps.get_model("zerver", "Message")
+    ArchivedMessage = apps.get_model("zerver", "ArchivedMessage")
+    ScheduledMessage = apps.get_model("zerver", "ScheduledMessage")
+
+    RECIPIENT_STREAM = 2
+
+    DM_TOPIC = "\x07"
+
+    BATCH_SIZE = 10000
+    for message_model in [Message, ArchivedMessage, ScheduledMessage]:
+        lower_bound = 1
+
+        max_id = message_model.objects.aggregate(Max("id"))["id__max"]
+        if max_id is None:
+            continue
+
+        while lower_bound <= max_id:
+            upper_bound = lower_bound + BATCH_SIZE - 1
+            print(f"Processing batch {lower_bound} to {upper_bound} for {message_model.__name__}")
+
+            message_model.objects.filter(
```

Comment:
Should we optimizie this queries specially for `Message` table which can have a very large number of rows?

Reply:
I don't see an obvious way to optimize this much. It's possible we could do better looping over `Recipient` as the outer loop, but I don't really expect that'd actually be a better query plan.